### PR TITLE
Renaming avm to nef

### PR DESCRIPTION
**Pull request by vncoelho**
closes https://github.com/neo-project/neo-compiler/issues/149

### CODE REVIEW

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-devpack-dotnet/pull/55

**Comment by vncoelho** [comment]
Perfect, my friend.
Is there any other place in which we need to rename? Or it is just `neo-cli` and dev-pack?

**Comment by shargon** [comment]
I don't know ... (readme and docs?)

**Comment by vncoelho** [comment]
That is true!

**Comment by vncoelho** [comment]
@shargon, please double check this last merge.

**Comment by erikzhang** [comment]
We need https://github.com/neo-project/neo/pull/903.

File: neo-cli/Shell/MainService.cs [hunk 1]
```diff
@@ -221,15 +221,15 @@ private bool OnInvokeCommand(string[] args)
...
-            var info = new FileInfo(nvmFilePath);
+            var info = new FileInfo(nefFilePath);
             if (!info.Exists || info.Length >= Transaction.MaxTransactionSize)
             {
-                throw new ArgumentException(nameof(nvmFilePath));
+                throw new ArgumentException(nameof(nefFilePath));
             }
 
-            byte[] script = File.ReadAllBytes(nvmFilePath);
+            byte[] script = File.ReadAllBytes(nefFilePath);
```
Comment (shargon) [inline neo-cli/Shell/MainService.cs]: We need to use the NefFile class

Comment (vncoelho) [reply neo-cli/Shell/MainService.cs]: @shargon, you mean get the `nefFilePath` and serialize the class and get the script from it, right?

Can you proceed with this change?

Comment (igormcoelho) [reply neo-cli/Shell/MainService.cs]: We just need to decide and count the NEF bytes... this is the header.

Comment (igormcoelho) [reply neo-cli/Shell/MainService.cs]: I guess they did a nice summary on other PR, Vitor:
```
    /// +------------+-----------+------------------------------------------------------------+
    /// |   Field    |  Length   |                          Comment                           |
    /// +------------+-----------+------------------------------------------------------------+
    /// | Magic      | 4 bytes   | Magic header                                               |
    /// | Compiler   | 32 bytes  | Compiler used                                              |
    /// | Version    | 16 bytes  | Compiler version (Mayor, Minor, Build, Version)            |
    /// | ScriptHash | 20 bytes  | ScriptHash for the script                                  |
    /// +------------+-----------+------------------------------------------------------------+
    /// | Checksum   | 4 bytes   | Sha256 of the header (CRC)                                 |
    /// +------------+-----------+------------------------------------------------------------+
    /// | Script     | Var bytes | Var bytes for the payload                                  |
    /// +------------+-----------+------------------------------------------------------------+
```
I think it's 76bytes + script... on script, it's important to load it as ReadVarBytes, so its "header" (byte count) is also automatically removed.

**Pull request merged**